pg-sharding/spqr
# Add Dataspace Task

Command change:
CREATE DATASPACE ds1 - create new dataspace with id = ds1
ADD SHARDING RULE r1 DATASPACE ds1 TABLE test1 COLUMNS age - add new sharding rule with link to ds1
ADD KEY RANGE krid11 FROM 1 TO 18 ROUTE TO sh1 DATASPACE ds1; - add new key range with link to ds1

old version commands(without dataspace) saved


to test, we can use this example:

ds1:
test1.age:
1-18 -> sh1
19-100 -> sh2
101-3000 -> sh3

ds2:
test1.age:
1-1000 -> sh1
1001 - 2000 -> sh2
2001 - 3000 -> sh3

*connect to test1 table, ds1 dataspace, db1 database*
INSERT INTO test1(age) VALUES(10)
INSERT INTO test1(age) VALUES(50)
INSERT INTO test1(age) VALUES(500)
INSERT INTO test1(age) VALUES(1500)
INSERT INTO test1(age) VALUES(2500)


*connect to test1 table, ds2 dataspace, db1 database*
INSERT INTO test1(age) VALUES(10)
INSERT INTO test1(age) VALUES(50)
INSERT INTO test1(age) VALUES(510)
INSERT INTO test1(age) VALUES(1500)
INSERT INTO test1(age) VALUES(2500)


in db should:
sh1:             sh2:                sh3:
10                50                  500
10                1500              1500
50                                      2500
500                                    2500
